### PR TITLE
Enhancement: Parsons Table: added rename_columns for multiple cols

### PR DESCRIPTION
**Pull request by mkrausse-ggtx**
This pull request introduces a new method, rename_columns, to the ETL process. This method allows for the renaming of multiple columns at once in a Parsons Table.

The rename_columns method takes a dictionary as an argument (column_map), where the keys represent the current column names and the values represent the new column names. The method checks if the old column name exists and if the new column name does not already exist in the table. If these conditions are met, the method uses the underlying petl.rename function to perform the renaming operation.

The method returns the updated Parsons Table and also updates the self.table attribute of the class instance.

Unit tests have been added to ensure the rename_columns method works as expected and handles edge cases appropriately. These tests cover scenarios such as renaming columns with a valid column map, renaming only some columns, and attempting to rename a column that does not exist or to a name that already exists.

This enhancement provides a more efficient way to rename multiple columns, improving the flexibility and usability of the ETL process.

### CODE REVIEW

File: parsons/etl/etl.py [hunk 1]
```diff
@@ -74,6 +74,29 @@ def rename_column(self, column_name, new_column_name):
 
         return self
 
+    def rename_columns(self, column_map):
+        """
+        Rename multiple columns
+
+        `Args:`
+            column_map: dict
+                A dictionary of columns and new names
```
Comment (shaunagm) [inline parsons/etl/etl.py:83]: Can you specify here that the old names are keys and the new names are values? ie `{'old_name': 'new_name'}`?

**Review by shaunagm — CHANGES_REQUESTED** [review]
Overall this looks good, just a couple small docs changes. One is an adjustment to your docstring, the other is to this file: https://github.com/move-coop/parsons/blob/main/docs/table.rst#parsons-table-transformations

You can see there's a table that lists available Parsons Table transformations. If you can add a row underneath rename_column for your new rename_columns method, that should allow people to find your new method more easily.

File: parsons/etl/etl.py (hunk 1 above)
Comment (mkrausse-ggtx) [reply parsons/etl/etl.py:83]: Sure things. Resolved.

**Comment by shaunagm** [comment]
Why did you close this? Do you need help updating your PR?

**Comment by mkrausse-ggtx** [comment]
> Why did you close this? Do you need help updating your PR?

Looks like when I tried to rename the branch to the naming convention I'm trying to keep, it closed it. Reopened.
